api/config/v1alpha1/shared_types.go

```diff
@@ -37,6 +43,12 @@ type KubernetesDeploymentSpec struct {
 	// +optional
 	PodAnnotations map[string]string `json:"podAnnotations,omitempty"`
 
+	// Compute Resources required by this container.
+	// Cannot be updated.
```

Comment:
since this issue of a restart applies to all fields within `EnvoyProxy` would prefer to add this warning at a top level https://github.com/envoyproxy/gateway/blob/f56eb9c1f8808e40d96b372f845ad5447d613cab/api/config/v1alpha1/envoyproxy_types.go#L31 rather than on every field. It doesnt need to be added in this PR though

Reply:
yeah I meant lets rm this line from here and we can add a top level warning/string

Reply:
OK